### ISSUE TITLE
Display database connection string in log to aid connection debugging

### ISSUE THREAD

**Issue by sldorman** [metabase/metabase#7443]
Many people have difficulty with database connections.  Some issues are due to the way they entered their information in the Admin panel, some are due to true issues with Metabase.  It would be very helpful if the log showed the connection string being used by Metabase.  The user can then adjust their form entries as needed or can determine if a problem exists in Metabase.  

Right now, the log simply shows the following for most SQL Server connection issues:

04-25 10:37:12 ←[1mERROR metabase.driver←[0m :: Failed to connect to database: Timed out after 5000 milliseconds.
04-25 10:37:12 ←[1mDEBUG metabase.middleware←[0m :: PUT /api/database/35 400 (5s) (1 DB calls).
{:valid false, :dbname "Timed out after 5000 milliseconds.", :message "Timed out after 5000 milliseconds."}

#1854 
#6056

**Comment by salsakran** [comment]
If we did this, it would definitely need to be without the password. Note that a number of people encrypt DB user/password/etc and spitting it out into the logs unencrypted would be a major danger of this.

**Comment by sldorman** [comment]
Good point.  Would be nice if there were different levels of logging so this would only appear if a "debug level" of logging is set, a level that would never be used in a normal set up.
